packages/core/src/components/essentials/legend.ts

```diff
@@ -31,6 +32,34 @@ export class Legend extends Component {
 		// Configs
 		const checkboxRadius = options.legend.checkbox.radius;
 
+		// Truncation
+		// get user provided custom values for truncation
+		let truncationType = Tools.getProperty(
+			legendOptions,
+			"truncation",
+			"type"
+		);
+		let truncationThreshold = Tools.getProperty(
+			legendOptions,
+			"truncation",
+			"threshold"
+		);
+		let truncationNumCharacter = Tools.getProperty(
+			legendOptions,
+			"truncation",
+			"numCharacter"
+		);
+		// load default config for truncation if not provided
+		if (!truncationType) {
+			truncationType = Configuration.legend.truncation.type;
+		}
+		if (!truncationThreshold) {
+			truncationThreshold = Configuration.legend.truncation.threshold;
+		}
+		if (!truncationNumCharacter) {
+			truncationNumCharacter = Configuration.legend.truncation.numCharacter;
+		}
+
```

Comment:
`legend` options that you've set defaults for in `configuration.ts` are part of the base chart options which are used in merging user-provided options with our defaults

Reply:
![image](https://user-images.githubusercontent.com/14989804/85416091-14367580-b53c-11ea-923c-1fbe2eb5fd39.png)

Reply:
so your default legend values will be there unless the user sets them to `null` or `undefined`, you don't need to set them manually

Reply:
Resolved. Thanks.